packages/teams-js/src/internal/appId.ts

```diff
@@ -0,0 +1,30 @@
+import { validateStringAsAppId } from './appIdValidation';
+
+/**
+ * A strongly-typed class used to represent a "valid" app id.
+ *
+ * Valid is a relative term, in this case. Truly valid app ids are UUIDs as documented in the schema:
+ * https://learn.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema#id
+ * However, there are some older internal/hard-coded apps which violate this schema and use names like
+ * com.microsoft.teamspace.tab.youtube. For compatibility with these legacy apps, we unfortunately cannot
```

Comment:
Feel free to ignore or otherwise backburner this, but if we had a complete set of what the ids are from telemetry we could probably check for them and then make the AppID validation for *other* appIds a lot more robust

Reply:
Created a bug to explore future opportunities here.